client/src/pages/Home/index.js

```diff
@@ -1,24 +1,65 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 
 import { Pagination, ResourceList } from "../../components";
 import { useSearchParams } from "../../hooks";
 import { ResourceService, useService } from "../../services";
 
-export function Home() {
+const Home = () => {
 	const resourceService = useService(ResourceService);
 	const searchParams = useSearchParams();
-	const [{ lastPage, resources } = {}, setEnvelope] = useState();
+	const [{ lastPage, resources } = {}, setEnvelope] = useState({});
+
+	const storedPageSize = localStorage.getItem("pageSizePreference");
+	let initialPageSize;
+
+	try {
+		initialPageSize = storedPageSize ? JSON.parse(storedPageSize) : "default";
+	} catch (error) {
+		// Handle the error as needed
+		initialPageSize = "default";
+		throw new Error("Error parsing page size preference:", error);
+	}
+
+	const [pageSize, setPageSize] = useState(initialPageSize);
+
+	// This function is to save the selected page-size into local storage
+	const savePageSizePreference = (pageSize) => {
+		localStorage.setItem("pageSizePreference", JSON.stringify(pageSize));
+	};
 
 	useEffect(() => {
-		resourceService.getPublished(searchParams).then(setEnvelope);
-	}, [resourceService, searchParams]);
+		// Using the selected page size in my API request
+		resourceService
+			.getPublished({ ...searchParams, pageSize })
+			.then(setEnvelope)
+			.catch((error) => {
+				// Handle the error from the API request as needed
+				throw new Error("Error fetching published resources:", error);
```

Comment:
❌  Throwing an error in this way is generally not helpful - if you want to created a new error "caused by" the previous one it should be:

```suggestion
				throw new Error("Error fetching published resources:", { cause: error });
```

[MDN][1]

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#rethrowing_an_error_with_a_cause

But if you can't really deal with the error or add any useful information, it's best to let it keep going to wherever it ends up.